File: testbed/test/cruise/associations/OneToOneTest.java
package cruise.associations;

import org.junit.*;

import cruise.associations.MentorG;
import cruise.associations.ProgramG;
import cruise.associations.StudentG;

public class OneToOneTest
{

  @Test
  public void ConstructorBuildsBoth()
  {
    MentorG m = new MentorG("a",1);
    Assert.assertEquals("a",m.getName());
    Assert.assertEquals(1, m.getStudent().getNumber());
  }
  
  @Test(expected=RuntimeException.class)
  public void ConstructorIfAlreadySet()
  {
    MentorG m = new MentorG("a",1);
    new StudentG(1,m);
  }  
  
  @Test(expected=RuntimeException.class)
  public void ConstructorCannotSetNull()
  {
    new MentorG("a",(StudentG)null);
  }   
  
  @Test
  public void delete()
  {
    MentorG m = new MentorG("a",1);
    StudentG s = m.getStudent();
    m.getStudent().setProgram(new ProgramG());
    m.delete();
    Assert.assertEquals(null, m.getStudent());
    Assert.assertEquals(null, m.getProgram());
    Assert.assertEquals(null, s.getMentor());
    Assert.assertEquals(null, s.getProgram());
    
  }
  
}

File: testbed_python/test/associations/one_to_one_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["MentorG", "ProgramG", "StudentG"], ["associations"])
from ImportModules import *


class OneToOneTest(unittest.TestCase):
    def test_constructorBuildsBoth(self):
        m = MentorG.MentorG("a", 1)
        self.assertEqual("a", m.getName())
        self.assertEqual(1, m.getStudent().getNumber())

    def test_constructorIfAlreadySet(self):
        with self.assertRaises(RuntimeError):
            m = MentorG.MentorG("a", 1)
            StudentG.StudentG.alternateConstructor(1, m)

    def test_constructorCannotSetNone(self):
        with self.assertRaises(RuntimeError):
            MentorG.MentorG.alternateConstructor("a", None)

    def test_delete(self):
        m = MentorG.MentorG("a", 1)
        s = m.getStudent()
        m.getStudent().setProgram(ProgramG.ProgramG())
        del m
        self.assertFalse(s.getMentor() in locals())
        self.assertFalse(s.getProgram() in locals())
